tests/integration/update_cluster/externalpolicies/kubernetes.tf

```diff
@@ -0,0 +1,644 @@
+    Name                                        = "us-test-1a.etcd-events.complex.example.com"
+    Owner                                       = "John Doe"
+    "foo/bar"                                   = "fib+baz"
+    "k8s.io/etcd/events"                        = "us-test-1a/us-test-1a"
+    "k8s.io/role/master"                        = "1"
+    "kubernetes.io/cluster/complex.example.com" = "owned"
+  }
+}
+
+resource "aws_ebs_volume" "us-test-1a-etcd-main-complex-example-com" {
+  availability_zone = "us-test-1a"
+  size              = 20
+  type              = "gp2"
+  encrypted         = false
+
+  tags = {
+    KubernetesCluster                           = "complex.example.com"
+    Name                                        = "us-test-1a.etcd-main.complex.example.com"
+    Owner                                       = "John Doe"
+    "foo/bar"                                   = "fib+baz"
+    "k8s.io/etcd/main"                          = "us-test-1a/us-test-1a"
+    "k8s.io/role/master"                        = "1"
+    "kubernetes.io/cluster/complex.example.com" = "owned"
+  }
+}
+
+resource "aws_elb" "api-complex-example-com" {
+  name = "api-complex-example-com-vd3t5n"
+
+  listener = {
+    instance_port     = 443
+    instance_protocol = "TCP"
+    lb_port           = 443
+    lb_protocol       = "TCP"
+  }
+
+  security_groups = ["${aws_security_group.api-elb-complex-example-com.id}", "sg-exampleid3", "sg-exampleid4"]
+  subnets         = ["${aws_subnet.us-test-1a-complex-example-com.id}"]
+
+  health_check = {
+    target              = "SSL:443"
+    healthy_threshold   = 2
+    unhealthy_threshold = 2
+    interval            = 10
+    timeout             = 5
+  }
+
+  cross_zone_load_balancing = false
+  idle_timeout              = 300
+
+  tags = {
+    KubernetesCluster                           = "complex.example.com"
+    Name                                        = "api.complex.example.com"
+    Owner                                       = "John Doe"
+    "foo/bar"                                   = "fib+baz"
+    "kubernetes.io/cluster/complex.example.com" = "owned"
+  }
+}
+
+resource "aws_iam_instance_profile" "masters-complex-example-com" {
+  name = "masters.complex.example.com"
+  role = "${aws_iam_role.masters-complex-example-com.name}"
+}
+
+resource "aws_iam_instance_profile" "nodes-complex-example-com" {
+  name = "nodes.complex.example.com"
+  role = "${aws_iam_role.nodes-complex-example-com.name}"
+}
+
+resource "aws_iam_role" "masters-complex-example-com" {
+  name               = "masters.complex.example.com"
+  assume_role_policy = "${file("${path.module}/data/aws_iam_role_masters.complex.example.com_policy")}"
+}
+
+resource "aws_iam_role" "nodes-complex-example-com" {
+  name               = "nodes.complex.example.com"
+  assume_role_policy = "${file("${path.module}/data/aws_iam_role_nodes.complex.example.com_policy")}"
+}
+
+resource "aws_iam_role_policy_attachment" "node-externalpolicy" {
```

Comment:
These terraform resources look much better but the resource name still seems questionable. Your code below:

```go
h := fnv.New32a()
h.Write([]byte(policy))

name := fmt.Sprintf("%s-%d", *e.Name, h.Sum32())
```

makes me expect this line to be more like:

```tf
resource "aws_iam_role_policy_attachment" "node-1178482798" {
```

which is definitely what we want. I'm having trouble figuring out where `externalpolicy` comes from exactly. Any ideas?

Reply:
Good catch, I just forgot to update the name in the .tf file

Reply:
Ah now I understand whats happening. We need to add code that actually executes this test to assert that the kubernetes.tf file you have here is what Kops generates.

A good starting point would be copying one of these functions:

https://github.com/kubernetes/kops/blob/4f2cddaa8b17ad07a444c5785bdc8ccd89bee622/cmd/kops/integration_test.go#L53-L58

(only call `runTestAWS` once with `v1alpha2` since thats the only cluster manifest file that you have in this new test)